typescript_typeorm_graphql_project/src/controllers/auth.ts

```diff
@@ -0,0 +1,104 @@
+import { validationResult } from "express-validator";
+import { hash, compare } from "bcryptjs";
+import { sign } from "jsonwebtoken";
+import * as dotenv from "dotenv";
+import { Request, Response, NextFunction } from "express";
+import { User } from "../entities/User";
+dotenv.config();
+
+const signUp = async (
+  req: Request | any,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const errors = validationResult(req);
+    console.log(req.body);
+
+    if (!errors.isEmpty()) {
+      res.statusCode = 422;
+      const error = new Error("Validation Failed.");
+      throw error;
+    }
+
+    const email: string = req.body.email;
+    const name: string = req.body.name;
+    const password: string = req.body.password;
+    const role: string = req.body.role;
+
+    const hashPassword = await hash(password, 12);
+
+    console.log(hashPassword);
+
+    const user = new User();
+
+    user.email = email;
+    user.name = name;
+    user.role = role;
+    user.password = hashPassword;
+
+    const result = await user.save();
+
+    res
+      .status(200)
+      .json({ message: "User created!", userId: result.id, userRole: role });
+    return result;
+  } catch (err: any) {
+    if (!err.statusCode) {
```

Comment:
no need to add additional check here for statuscode.
check is already present in error middleware and "/graphql" endpoint

`const status = error.statusCode || 500;
  const message = error.message;`

Reply:
Updated